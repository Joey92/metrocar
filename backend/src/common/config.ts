import fs from 'fs';
import * as Sentry from '@sentry/node';

if (process.env.SENTRY_DSN) {
  Sentry.init({
    dsn: process.env.SENTRY_DSN,

    // Set tracesSampleRate to 1.0 to capture 100%
    // of transactions for performance monitoring.
    // We recommend adjusting this value in production
    tracesSampleRate: 1.0,
    environment: process.env.NODE_ENV,
  });
}

let jwtKey;

if (!process.env.JWT_PRIVATE_KEY?.startsWith('/run/secrets')) {
  // URL directly passed
  console.log('using jwt private key from env');
  jwtKey = process.env.JWT_PRIVATE_KEY;
} else {
  // path to file/secret
  console.log('using jwt private key from file');
  jwtKey = fs.readFileSync(process.env.JWT_PRIVATE_KEY, { encoding: 'utf8' });
}

export const jwtVerifyKey = jwtKey;

export const POSTGRES_URL =
  process.env.POSTGRES_URL || 'postgres://root:root@127.0.0.1/metrocar';

if (POSTGRES_URL === '') {
  throw new Error('POSTGRES_URL needs to be set in the environment!');
}

let postgres = '';

if (POSTGRES_URL.startsWith('postgres://')) {
  // URL directly passed
  postgres = POSTGRES_URL;
} else {
  // path to file/secret
  postgres = fs.readFileSync(POSTGRES_URL, 'utf8');
}

export const frontendUrl = process.env.FRONTEND_URL || 'https://metrocar.org';
export const postgresUrl = postgres;

export const REDIS_URL = process.env.REDIS_URL || 'redis://127.0.0.1:6380/1';

export const GRAPHHOPPER_URL =
  process.env.GRAPHHOPPER_URL || 'http://127.0.0.1:8989';

export const OTP_URL = process.env.OTP_URL || 'http://127.0.0.1:8082';

export const NOMINATIM_URL =
  process.env.NOMINATIM_URL || 'http://127.0.0.1:8083';

export const PHOTON_URL = process.env.PHOTON_URL || 'photon';

export const DIRECTUS_URL = process.env.DIRECTUS_URL;
export const DIRECTUS_USER_TOKEN = process.env.DIRECTUS_USER_TOKEN;

export const INTERNAL_DEFAULT_AGENCY_ID =
  process.env.INTERNAL_DEFAULT_AGENCY_ID;
