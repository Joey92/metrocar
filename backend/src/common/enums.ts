export enum Messages {
  USER_EXISTS = 'user_exists',
  FIELD_REQUIRED = 'field_required',
  FIELD_INVALID = 'field_invalid',
  ROUTER_UNAVAILABLE = 'router_unavailable',
  REVERSE_GEOCODE_UNAVAILABLE = 'reverse_geocode_unavailable',

  NO_STOPS_ON_ROUTE = 'no_stops_on_route',

  INTERNAL_SERVER_ERROR = 'internal_server_error',

  PLACE_NOT_FOUND = 'place_not_found',

  USERNAME_INVALID = 'username_invalid',
  EMAIL_INVALID = 'email_invalid',
  EMAIL_REGISTERED = 'email_registered',

  ACCOUNT_CREATED = 'account_created',

  LOGIN_FAILED = 'login_failed',

  GENERIC_ERROR = 'generic_error',
}
