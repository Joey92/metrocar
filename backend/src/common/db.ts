import pgp from 'pg-promise';
import * as config from './config';

export const pg = pgp();
export const db = pg({
  max: 200,
  connectionString: config.postgresUrl,
});
