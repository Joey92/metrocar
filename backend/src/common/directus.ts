import axios from 'axios';
import { DIRECTUS_URL, DIRECTUS_USER_TOKEN } from './config';

export const directus = axios.create({
  baseURL: DIRECTUS_URL,
  headers: {
    Authorization: `Bearer ${DIRECTUS_USER_TOKEN}`,
  },
});

export default directus;
