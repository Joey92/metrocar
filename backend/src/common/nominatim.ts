import axios from 'axios';
import * as config from './config';

export interface NominatimResponseElement {
  lat: string;
  lon: string;
  display_name: string;
}

export const nominatimRequest = (q: string) =>
  axios.get<NominatimResponseElement[]>(`${config.NOMINATIM_URL}/search`, {
    params: { q },
  });
