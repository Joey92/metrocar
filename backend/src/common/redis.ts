import { REDIS_URL } from './config';
import { createClient } from 'redis';

export const redisClient = createClient({
  url: REDIS_URL,
});

redisClient.on('error', (err) => console.log('Redis Client Error', err));

redisClient.connect().catch((e) => {
  throw e;
});

export default redisClient;
