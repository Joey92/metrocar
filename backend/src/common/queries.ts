import { IDatabase, ITask } from 'pg-promise';
import { StopData } from '../types';
import { pg, db } from './db';

export async function createPrivateStops(
  stops: StopData[],
  dbConn: ITask<any> | IDatabase<any> = db,
) {
  const stopInsert = pg.helpers.insert(
    stops.map((s) => ({ ...s, created_at: new Date() })),
    ['id', 'stop_name', 'location', 'reporter', 'private', 'created_at'],
    'stops',
  );
  return dbConn.none(stopInsert);
}
