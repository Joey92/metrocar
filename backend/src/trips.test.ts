import { db } from './common/db';
import { getGTFSShapeFromPoints } from './trips';
import { deleteTestDataQuery, testDataQuery } from './test_helper/helpers';

beforeAll(() => db.none(testDataQuery));
afterAll(() => db.none(deleteTestDataQuery));

it('makes shape correctly', async () => {
  const points = [
    [13.576265, 52.579501],
    [13.424863815307619, 52.52379372545992],
    [13.372711, 52.527128],
  ];

  const data = await getGTFSShapeFromPoints(points);

  expect(data).toMatchSnapshot();
});
