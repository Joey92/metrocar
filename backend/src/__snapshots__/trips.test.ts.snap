// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makes shape correctly 1`] = `
Array [
  Object {
    "shape_dist_traveled": 0,
    "shape_pt_lat": 13.576254,
    "shape_pt_lon": 52.579519,
    "shape_pt_sequence": 0,
  },
  Object {
    "shape_dist_traveled": 81,
    "shape_pt_lat": 13.575543,
    "shape_pt_lon": 52.579344,
    "shape_pt_sequence": 1,
  },
  Object {
    "shape_dist_traveled": 147,
    "shape_pt_lat": 13.575191,
    "shape_pt_lon": 52.579835,
    "shape_pt_sequence": 2,
  },
  Object {
    "shape_dist_traveled": 207,
    "shape_pt_lat": 13.574901,
    "shape_pt_lon": 52.580301,
    "shape_pt_sequence": 3,
  },
  Object {
    "shape_dist_traveled": 271,
    "shape_pt_lat": 13.574544,
    "shape_pt_lon": 52.580766,
    "shape_pt_sequence": 4,
  },
  Object {
    "shape_dist_traveled": 311,
    "shape_pt_lat": 13.574362,
    "shape_pt_lon": 52.581084,
    "shape_pt_sequence": 5,
  },
  Object {
    "shape_dist_traveled": 351,
    "shape_pt_lat": 13.574008,
    "shape_pt_lon": 52.581013,
    "shape_pt_sequence": 6,
  },
  Object {
    "shape_dist_traveled": 377,
    "shape_pt_lat": 13.573786,
    "shape_pt_lon": 52.580944,
    "shape_pt_sequence": 7,
  },
  Object {
    "shape_dist_traveled": 395,
    "shape_pt_lat": 13.57364,
    "shape_pt_lon": 52.580865,
    "shape_pt_sequence": 8,
  },
  Object {
    "shape_dist_traveled": 413,
    "shape_pt_lat": 13.573508,
    "shape_pt_lon": 52.580768,
    "shape_pt_sequence": 9,
  },
  Object {
    "shape_dist_traveled": 569,
    "shape_pt_lat": 13.572572,
    "shape_pt_lon": 52.579692,
    "shape_pt_sequence": 10,
  },
  Object {
    "shape_dist_traveled": 594,
    "shape_pt_lat": 13.572454,
    "shape_pt_lon": 52.579494,
    "shape_pt_sequence": 11,
  },
  Object {
    "shape_dist_traveled": 609,
    "shape_pt_lat": 13.572413,
    "shape_pt_lon": 52.579364,
    "shape_pt_sequence": 12,
  },
  Object {
    "shape_dist_traveled": 655,
    "shape_pt_lat": 13.572137,
    "shape_pt_lon": 52.579049,
    "shape_pt_sequence": 13,
  },
  Object {
    "shape_dist_traveled": 822,
    "shape_pt_lat": 13.573357,
    "shape_pt_lon": 52.578143,
    "shape_pt_sequence": 14,
  },
  Object {
    "shape_dist_traveled": 834,
    "shape_pt_lat": 13.573424,
    "shape_pt_lon": 52.578051,
    "shape_pt_sequence": 15,
  },
  Object {
    "shape_dist_traveled": 847,
    "shape_pt_lat": 13.573521,
    "shape_pt_lon": 52.577979,
    "shape_pt_sequence": 16,
  },
  Object {
    "shape_dist_traveled": 865,
    "shape_pt_lat": 13.573667,
    "shape_pt_lon": 52.577916,
    "shape_pt_sequence": 17,
  },
  Object {
    "shape_dist_traveled": 1038,
    "shape_pt_lat": 13.574929,
    "shape_pt_lon": 52.576983,
    "shape_pt_sequence": 18,
  },
  Object {
    "shape_dist_traveled": 1080,
    "shape_pt_lat": 13.575221,
    "shape_pt_lon": 52.576731,
    "shape_pt_sequence": 19,
  },
  Object {
    "shape_dist_traveled": 1241,
    "shape_pt_lat": 13.576215,
    "shape_pt_lon": 52.575652,
    "shape_pt_sequence": 20,
  },
  Object {
    "shape_dist_traveled": 1286,
    "shape_pt_lat": 13.576454,
    "shape_pt_lon": 52.575314,
    "shape_pt_sequence": 21,
  },
  Object {
    "shape_dist_traveled": 1379,
    "shape_pt_lat": 13.575634,
    "shape_pt_lon": 52.575167,
    "shape_pt_sequence": 22,
  },
  Object {
    "shape_dist_traveled": 1542,
    "shape_pt_lat": 13.574179,
    "shape_pt_lon": 52.574958,
    "shape_pt_sequence": 23,
  },
  Object {
    "shape_dist_traveled": 1729,
    "shape_pt_lat": 13.572515,
    "shape_pt_lon": 52.574689,
    "shape_pt_sequence": 24,
  },
  Object {
    "shape_dist_traveled": 1890,
    "shape_pt_lat": 13.571079,
    "shape_pt_lon": 52.574493,
    "shape_pt_sequence": 25,
  },
  Object {
    "shape_dist_traveled": 1964,
    "shape_pt_lat": 13.570423,
    "shape_pt_lon": 52.574382,
    "shape_pt_sequence": 26,
  },
  Object {
    "shape_dist_traveled": 1998,
    "shape_pt_lat": 13.570122,
    "shape_pt_lon": 52.57432,
    "shape_pt_sequence": 27,
  },
  Object {
    "shape_dist_traveled": 2079,
    "shape_pt_lat": 13.569421,
    "shape_pt_lon": 52.574103,
    "shape_pt_sequence": 28,
  },
  Object {
    "shape_dist_traveled": 2235,
    "shape_pt_lat": 13.568095,
    "shape_pt_lon": 52.573625,
    "shape_pt_sequence": 29,
  },
  Object {
    "shape_dist_traveled": 2288,
    "shape_pt_lat": 13.567809,
    "shape_pt_lon": 52.573229,
    "shape_pt_sequence": 30,
  },
  Object {
    "shape_dist_traveled": 2448,
    "shape_pt_lat": 13.566908,
    "shape_pt_lon": 52.572071,
    "shape_pt_sequence": 31,
  },
  Object {
    "shape_dist_traveled": 3080,
    "shape_pt_lat": 13.563079,
    "shape_pt_lon": 52.567751,
    "shape_pt_sequence": 32,
  },
  Object {
    "shape_dist_traveled": 3829,
    "shape_pt_lat": 13.558283,
    "shape_pt_lon": 52.562885,
    "shape_pt_sequence": 33,
  },
  Object {
    "shape_dist_traveled": 3855,
    "shape_pt_lat": 13.558124,
    "shape_pt_lon": 52.562708,
    "shape_pt_sequence": 34,
  },
  Object {
    "shape_dist_traveled": 4119,
    "shape_pt_lat": 13.556764,
    "shape_pt_lon": 52.56071,
    "shape_pt_sequence": 35,
  },
  Object {
    "shape_dist_traveled": 5032,
    "shape_pt_lat": 13.55132,
    "shape_pt_lon": 52.554389,
    "shape_pt_sequence": 36,
  },
  Object {
    "shape_dist_traveled": 5430,
    "shape_pt_lat": 13.54894,
    "shape_pt_lon": 52.55164,
    "shape_pt_sequence": 37,
  },
  Object {
    "shape_dist_traveled": 5480,
    "shape_pt_lat": 13.548629,
    "shape_pt_lon": 52.551305,
    "shape_pt_sequence": 38,
  },
  Object {
    "shape_dist_traveled": 6343,
    "shape_pt_lat": 13.54353,
    "shape_pt_lon": 52.545291,
    "shape_pt_sequence": 39,
  },
  Object {
    "shape_dist_traveled": 6754,
    "shape_pt_lat": 13.541164,
    "shape_pt_lon": 52.542375,
    "shape_pt_sequence": 40,
  },
  Object {
    "shape_dist_traveled": 6791,
    "shape_pt_lat": 13.540929,
    "shape_pt_lon": 52.542137,
    "shape_pt_sequence": 41,
  },
  Object {
    "shape_dist_traveled": 6833,
    "shape_pt_lat": 13.54069,
    "shape_pt_lon": 52.541841,
    "shape_pt_sequence": 42,
  },
  Object {
    "shape_dist_traveled": 7085,
    "shape_pt_lat": 13.539149,
    "shape_pt_lon": 52.540135,
    "shape_pt_sequence": 43,
  },
  Object {
    "shape_dist_traveled": 7124,
    "shape_pt_lat": 13.538848,
    "shape_pt_lon": 52.539947,
    "shape_pt_sequence": 44,
  },
  Object {
    "shape_dist_traveled": 7282,
    "shape_pt_lat": 13.537445,
    "shape_pt_lon": 52.539731,
    "shape_pt_sequence": 45,
  },
  Object {
    "shape_dist_traveled": 7348,
    "shape_pt_lat": 13.536859,
    "shape_pt_lon": 52.539626,
    "shape_pt_sequence": 46,
  },
  Object {
    "shape_dist_traveled": 7465,
    "shape_pt_lat": 13.535832,
    "shape_pt_lon": 52.539394,
    "shape_pt_sequence": 47,
  },
  Object {
    "shape_dist_traveled": 7618,
    "shape_pt_lat": 13.534505,
    "shape_pt_lon": 52.539023,
    "shape_pt_sequence": 48,
  },
  Object {
    "shape_dist_traveled": 7712,
    "shape_pt_lat": 13.533675,
    "shape_pt_lon": 52.53886,
    "shape_pt_sequence": 49,
  },
  Object {
    "shape_dist_traveled": 8032,
    "shape_pt_lat": 13.53086,
    "shape_pt_lon": 52.538243,
    "shape_pt_sequence": 50,
  },
  Object {
    "shape_dist_traveled": 8346,
    "shape_pt_lat": 13.528119,
    "shape_pt_lon": 52.537528,
    "shape_pt_sequence": 51,
  },
  Object {
    "shape_dist_traveled": 8867,
    "shape_pt_lat": 13.523625,
    "shape_pt_lon": 52.536172,
    "shape_pt_sequence": 52,
  },
  Object {
    "shape_dist_traveled": 8945,
    "shape_pt_lat": 13.52294,
    "shape_pt_lon": 52.536016,
    "shape_pt_sequence": 53,
  },
  Object {
    "shape_dist_traveled": 9052,
    "shape_pt_lat": 13.521988,
    "shape_pt_lon": 52.535848,
    "shape_pt_sequence": 54,
  },
  Object {
    "shape_dist_traveled": 9145,
    "shape_pt_lat": 13.521156,
    "shape_pt_lon": 52.535744,
    "shape_pt_sequence": 55,
  },
  Object {
    "shape_dist_traveled": 9553,
    "shape_pt_lat": 13.517494,
    "shape_pt_lon": 52.535481,
    "shape_pt_sequence": 56,
  },
  Object {
    "shape_dist_traveled": 9920,
    "shape_pt_lat": 13.514204,
    "shape_pt_lon": 52.535268,
    "shape_pt_sequence": 57,
  },
  Object {
    "shape_dist_traveled": 10406,
    "shape_pt_lat": 13.509845,
    "shape_pt_lon": 52.534961,
    "shape_pt_sequence": 58,
  },
  Object {
    "shape_dist_traveled": 10444,
    "shape_pt_lat": 13.509505,
    "shape_pt_lon": 52.534954,
    "shape_pt_sequence": 59,
  },
  Object {
    "shape_dist_traveled": 10698,
    "shape_pt_lat": 13.507221,
    "shape_pt_lon": 52.534825,
    "shape_pt_sequence": 60,
  },
  Object {
    "shape_dist_traveled": 11346,
    "shape_pt_lat": 13.501405,
    "shape_pt_lon": 52.534451,
    "shape_pt_sequence": 61,
  },
  Object {
    "shape_dist_traveled": 11704,
    "shape_pt_lat": 13.498186,
    "shape_pt_lon": 52.534268,
    "shape_pt_sequence": 62,
  },
  Object {
    "shape_dist_traveled": 11961,
    "shape_pt_lat": 13.495874,
    "shape_pt_lon": 52.534223,
    "shape_pt_sequence": 63,
  },
  Object {
    "shape_dist_traveled": 12093,
    "shape_pt_lat": 13.494685,
    "shape_pt_lon": 52.534188,
    "shape_pt_sequence": 64,
  },
  Object {
    "shape_dist_traveled": 12249,
    "shape_pt_lat": 13.493279,
    "shape_pt_lon": 52.534123,
    "shape_pt_sequence": 65,
  },
  Object {
    "shape_dist_traveled": 12930,
    "shape_pt_lat": 13.487166,
    "shape_pt_lon": 52.533778,
    "shape_pt_sequence": 66,
  },
  Object {
    "shape_dist_traveled": 12993,
    "shape_pt_lat": 13.486602,
    "shape_pt_lon": 52.53372,
    "shape_pt_sequence": 67,
  },
  Object {
    "shape_dist_traveled": 13167,
    "shape_pt_lat": 13.485042,
    "shape_pt_lon": 52.533626,
    "shape_pt_sequence": 68,
  },
  Object {
    "shape_dist_traveled": 13609,
    "shape_pt_lat": 13.48107,
    "shape_pt_lon": 52.533422,
    "shape_pt_sequence": 69,
  },
  Object {
    "shape_dist_traveled": 14137,
    "shape_pt_lat": 13.476329,
    "shape_pt_lon": 52.533096,
    "shape_pt_sequence": 70,
  },
  Object {
    "shape_dist_traveled": 14177,
    "shape_pt_lat": 13.475975,
    "shape_pt_lon": 52.533043,
    "shape_pt_sequence": 71,
  },
  Object {
    "shape_dist_traveled": 14670,
    "shape_pt_lat": 13.471636,
    "shape_pt_lon": 52.532098,
    "shape_pt_sequence": 72,
  },
  Object {
    "shape_dist_traveled": 15346,
    "shape_pt_lat": 13.465693,
    "shape_pt_lon": 52.530762,
    "shape_pt_sequence": 73,
  },
  Object {
    "shape_dist_traveled": 15535,
    "shape_pt_lat": 13.464026,
    "shape_pt_lon": 52.530403,
    "shape_pt_sequence": 74,
  },
  Object {
    "shape_dist_traveled": 15783,
    "shape_pt_lat": 13.461853,
    "shape_pt_lon": 52.529892,
    "shape_pt_sequence": 75,
  },
  Object {
    "shape_dist_traveled": 15936,
    "shape_pt_lat": 13.460503,
    "shape_pt_lon": 52.529601,
    "shape_pt_sequence": 76,
  },
  Object {
    "shape_dist_traveled": 16441,
    "shape_pt_lat": 13.456049,
    "shape_pt_lon": 52.528704,
    "shape_pt_sequence": 77,
  },
  Object {
    "shape_dist_traveled": 16561,
    "shape_pt_lat": 13.454992,
    "shape_pt_lon": 52.528471,
    "shape_pt_sequence": 78,
  },
  Object {
    "shape_dist_traveled": 16599,
    "shape_pt_lat": 13.454662,
    "shape_pt_lon": 52.52838,
    "shape_pt_sequence": 79,
  },
  Object {
    "shape_dist_traveled": 16663,
    "shape_pt_lat": 13.454122,
    "shape_pt_lon": 52.528174,
    "shape_pt_sequence": 80,
  },
  Object {
    "shape_dist_traveled": 16698,
    "shape_pt_lat": 13.453819,
    "shape_pt_lon": 52.528079,
    "shape_pt_sequence": 81,
  },
  Object {
    "shape_dist_traveled": 16891,
    "shape_pt_lat": 13.452121,
    "shape_pt_lon": 52.527693,
    "shape_pt_sequence": 82,
  },
  Object {
    "shape_dist_traveled": 17049,
    "shape_pt_lat": 13.450741,
    "shape_pt_lon": 52.527332,
    "shape_pt_sequence": 83,
  },
  Object {
    "shape_dist_traveled": 17391,
    "shape_pt_lat": 13.447766,
    "shape_pt_lon": 52.526521,
    "shape_pt_sequence": 84,
  },
  Object {
    "shape_dist_traveled": 17429,
    "shape_pt_lat": 13.447444,
    "shape_pt_lon": 52.526411,
    "shape_pt_sequence": 85,
  },
  Object {
    "shape_dist_traveled": 17672,
    "shape_pt_lat": 13.445344,
    "shape_pt_lon": 52.525773,
    "shape_pt_sequence": 86,
  },
  Object {
    "shape_dist_traveled": 17746,
    "shape_pt_lat": 13.444716,
    "shape_pt_lon": 52.525556,
    "shape_pt_sequence": 87,
  },
  Object {
    "shape_dist_traveled": 18008,
    "shape_pt_lat": 13.442448,
    "shape_pt_lon": 52.524894,
    "shape_pt_sequence": 88,
  },
  Object {
    "shape_dist_traveled": 18500,
    "shape_pt_lat": 13.438182,
    "shape_pt_lon": 52.523685,
    "shape_pt_sequence": 89,
  },
  Object {
    "shape_dist_traveled": 18775,
    "shape_pt_lat": 13.435731,
    "shape_pt_lon": 52.523372,
    "shape_pt_sequence": 90,
  },
  Object {
    "shape_dist_traveled": 18919,
    "shape_pt_lat": 13.434436,
    "shape_pt_lon": 52.523316,
    "shape_pt_sequence": 91,
  },
  Object {
    "shape_dist_traveled": 19095,
    "shape_pt_lat": 13.432854,
    "shape_pt_lon": 52.523339,
    "shape_pt_sequence": 92,
  },
  Object {
    "shape_dist_traveled": 19339,
    "shape_pt_lat": 13.430657,
    "shape_pt_lon": 52.523303,
    "shape_pt_sequence": 93,
  },
  Object {
    "shape_dist_traveled": 19720,
    "shape_pt_lat": 13.427236,
    "shape_pt_lon": 52.523209,
    "shape_pt_sequence": 94,
  },
  Object {
    "shape_dist_traveled": 19816,
    "shape_pt_lat": 13.426375,
    "shape_pt_lon": 52.523202,
    "shape_pt_sequence": 95,
  },
  Object {
    "shape_dist_traveled": 19862,
    "shape_pt_lat": 13.425961,
    "shape_pt_lon": 52.523239,
    "shape_pt_sequence": 96,
  },
  Object {
    "shape_dist_traveled": 19917,
    "shape_pt_lat": 13.425474,
    "shape_pt_lon": 52.523337,
    "shape_pt_sequence": 97,
  },
  Object {
    "shape_dist_traveled": 20008,
    "shape_pt_lat": 13.424685,
    "shape_pt_lon": 52.523571,
    "shape_pt_sequence": 98,
  },
  Object {
    "shape_dist_traveled": 20082,
    "shape_pt_lat": 13.424044,
    "shape_pt_lon": 52.523761,
    "shape_pt_sequence": 99,
  },
  Object {
    "shape_dist_traveled": 20174,
    "shape_pt_lat": 13.423256,
    "shape_pt_lon": 52.524031,
    "shape_pt_sequence": 100,
  },
  Object {
    "shape_dist_traveled": 20441,
    "shape_pt_lat": 13.421009,
    "shape_pt_lon": 52.524908,
    "shape_pt_sequence": 101,
  },
  Object {
    "shape_dist_traveled": 20800,
    "shape_pt_lat": 13.418016,
    "shape_pt_lon": 52.526142,
    "shape_pt_sequence": 102,
  },
  Object {
    "shape_dist_traveled": 21050,
    "shape_pt_lat": 13.415945,
    "shape_pt_lon": 52.527047,
    "shape_pt_sequence": 103,
  },
  Object {
    "shape_dist_traveled": 21082,
    "shape_pt_lat": 13.415682,
    "shape_pt_lon": 52.527177,
    "shape_pt_sequence": 104,
  },
  Object {
    "shape_dist_traveled": 21139,
    "shape_pt_lat": 13.415217,
    "shape_pt_lon": 52.527401,
    "shape_pt_sequence": 105,
  },
  Object {
    "shape_dist_traveled": 21154,
    "shape_pt_lat": 13.415093,
    "shape_pt_lon": 52.527443,
    "shape_pt_sequence": 106,
  },
  Object {
    "shape_dist_traveled": 21436,
    "shape_pt_lat": 13.412648,
    "shape_pt_lon": 52.528134,
    "shape_pt_sequence": 107,
  },
  Object {
    "shape_dist_traveled": 21510,
    "shape_pt_lat": 13.41201,
    "shape_pt_lon": 52.528339,
    "shape_pt_sequence": 108,
  },
  Object {
    "shape_dist_traveled": 21578,
    "shape_pt_lat": 13.411418,
    "shape_pt_lon": 52.528507,
    "shape_pt_sequence": 109,
  },
  Object {
    "shape_dist_traveled": 21650,
    "shape_pt_lat": 13.410778,
    "shape_pt_lon": 52.52863,
    "shape_pt_sequence": 110,
  },
  Object {
    "shape_dist_traveled": 21793,
    "shape_pt_lat": 13.409508,
    "shape_pt_lon": 52.528838,
    "shape_pt_sequence": 111,
  },
  Object {
    "shape_dist_traveled": 21938,
    "shape_pt_lat": 13.408228,
    "shape_pt_lon": 52.529109,
    "shape_pt_sequence": 112,
  },
  Object {
    "shape_dist_traveled": 22256,
    "shape_pt_lat": 13.40541,
    "shape_pt_lon": 52.529597,
    "shape_pt_sequence": 113,
  },
  Object {
    "shape_dist_traveled": 22436,
    "shape_pt_lat": 13.403812,
    "shape_pt_lon": 52.529858,
    "shape_pt_sequence": 114,
  },
  Object {
    "shape_dist_traveled": 22506,
    "shape_pt_lat": 13.403182,
    "shape_pt_lon": 52.529882,
    "shape_pt_sequence": 115,
  },
  Object {
    "shape_dist_traveled": 22670,
    "shape_pt_lat": 13.401707,
    "shape_pt_lon": 52.529817,
    "shape_pt_sequence": 116,
  },
  Object {
    "shape_dist_traveled": 22720,
    "shape_pt_lat": 13.401265,
    "shape_pt_lon": 52.529752,
    "shape_pt_sequence": 117,
  },
  Object {
    "shape_dist_traveled": 22754,
    "shape_pt_lat": 13.400964,
    "shape_pt_lon": 52.529676,
    "shape_pt_sequence": 118,
  },
  Object {
    "shape_dist_traveled": 22770,
    "shape_pt_lat": 13.400834,
    "shape_pt_lon": 52.529612,
    "shape_pt_sequence": 119,
  },
  Object {
    "shape_dist_traveled": 23134,
    "shape_pt_lat": 13.397564,
    "shape_pt_lon": 52.529429,
    "shape_pt_sequence": 120,
  },
  Object {
    "shape_dist_traveled": 23344,
    "shape_pt_lat": 13.395723,
    "shape_pt_lon": 52.528995,
    "shape_pt_sequence": 121,
  },
  Object {
    "shape_dist_traveled": 23562,
    "shape_pt_lat": 13.393814,
    "shape_pt_lon": 52.528526,
    "shape_pt_sequence": 122,
  },
  Object {
    "shape_dist_traveled": 24134,
    "shape_pt_lat": 13.389731,
    "shape_pt_lon": 52.531741,
    "shape_pt_sequence": 123,
  },
  Object {
    "shape_dist_traveled": 24419,
    "shape_pt_lat": 13.387174,
    "shape_pt_lon": 52.531504,
    "shape_pt_sequence": 124,
  },
  Object {
    "shape_dist_traveled": 24462,
    "shape_pt_lat": 13.386787,
    "shape_pt_lon": 52.53151,
    "shape_pt_sequence": 125,
  },
  Object {
    "shape_dist_traveled": 24504,
    "shape_pt_lat": 13.386413,
    "shape_pt_lon": 52.531474,
    "shape_pt_sequence": 126,
  },
  Object {
    "shape_dist_traveled": 24557,
    "shape_pt_lat": 13.38594,
    "shape_pt_lon": 52.531392,
    "shape_pt_sequence": 127,
  },
  Object {
    "shape_dist_traveled": 24595,
    "shape_pt_lat": 13.385607,
    "shape_pt_lon": 52.531299,
    "shape_pt_sequence": 128,
  },
  Object {
    "shape_dist_traveled": 24630,
    "shape_pt_lat": 13.385313,
    "shape_pt_lon": 52.531179,
    "shape_pt_sequence": 129,
  },
  Object {
    "shape_dist_traveled": 24809,
    "shape_pt_lat": 13.383755,
    "shape_pt_lon": 52.530747,
    "shape_pt_sequence": 130,
  },
  Object {
    "shape_dist_traveled": 24888,
    "shape_pt_lat": 13.383071,
    "shape_pt_lon": 52.530567,
    "shape_pt_sequence": 131,
  },
  Object {
    "shape_dist_traveled": 24903,
    "shape_pt_lat": 13.382935,
    "shape_pt_lon": 52.53058,
    "shape_pt_sequence": 132,
  },
  Object {
    "shape_dist_traveled": 24922,
    "shape_pt_lat": 13.382764,
    "shape_pt_lon": 52.530564,
    "shape_pt_sequence": 133,
  },
  Object {
    "shape_dist_traveled": 25041,
    "shape_pt_lat": 13.381736,
    "shape_pt_lon": 52.530268,
    "shape_pt_sequence": 134,
  },
  Object {
    "shape_dist_traveled": 25069,
    "shape_pt_lat": 13.381512,
    "shape_pt_lon": 52.530149,
    "shape_pt_sequence": 135,
  },
  Object {
    "shape_dist_traveled": 25293,
    "shape_pt_lat": 13.379574,
    "shape_pt_lon": 52.529575,
    "shape_pt_sequence": 136,
  },
  Object {
    "shape_dist_traveled": 25344,
    "shape_pt_lat": 13.379141,
    "shape_pt_lon": 52.529409,
    "shape_pt_sequence": 137,
  },
  Object {
    "shape_dist_traveled": 25425,
    "shape_pt_lat": 13.378445,
    "shape_pt_lon": 52.529199,
    "shape_pt_sequence": 138,
  },
  Object {
    "shape_dist_traveled": 25451,
    "shape_pt_lat": 13.37821,
    "shape_pt_lon": 52.529186,
    "shape_pt_sequence": 139,
  },
  Object {
    "shape_dist_traveled": 25496,
    "shape_pt_lat": 13.377816,
    "shape_pt_lon": 52.529075,
    "shape_pt_sequence": 140,
  },
  Object {
    "shape_dist_traveled": 25567,
    "shape_pt_lat": 13.377195,
    "shape_pt_lon": 52.528939,
    "shape_pt_sequence": 141,
  },
  Object {
    "shape_dist_traveled": 25686,
    "shape_pt_lat": 13.376169,
    "shape_pt_lon": 52.528636,
    "shape_pt_sequence": 142,
  },
  Object {
    "shape_dist_traveled": 25715,
    "shape_pt_lat": 13.37596,
    "shape_pt_lon": 52.528472,
    "shape_pt_sequence": 143,
  },
  Object {
    "shape_dist_traveled": 25902,
    "shape_pt_lat": 13.374347,
    "shape_pt_lon": 52.527977,
    "shape_pt_sequence": 144,
  },
  Object {
    "shape_dist_traveled": 25937,
    "shape_pt_lat": 13.374033,
    "shape_pt_lon": 52.527946,
    "shape_pt_sequence": 145,
  },
  Object {
    "shape_dist_traveled": 26278,
    "shape_pt_lat": 13.37109,
    "shape_pt_lon": 52.527051,
    "shape_pt_sequence": 146,
  },
  Object {
    "shape_dist_traveled": 26355,
    "shape_pt_lat": 13.370417,
    "shape_pt_lon": 52.526897,
    "shape_pt_sequence": 147,
  },
  Object {
    "shape_dist_traveled": 26382,
    "shape_pt_lat": 13.370568,
    "shape_pt_lon": 52.526706,
    "shape_pt_sequence": 148,
  },
  Object {
    "shape_dist_traveled": 26432,
    "shape_pt_lat": 13.370982,
    "shape_pt_lon": 52.526875,
    "shape_pt_sequence": 149,
  },
  Object {
    "shape_dist_traveled": 26614,
    "shape_pt_lat": 13.372549,
    "shape_pt_lon": 52.527367,
    "shape_pt_sequence": 150,
  },
  Object {
    "shape_dist_traveled": 26635,
    "shape_pt_lat": 13.372681,
    "shape_pt_lon": 52.527228,
    "shape_pt_sequence": 151,
  },
  Object {
    "shape_dist_traveled": 26646,
    "shape_pt_lat": 13.372682,
    "shape_pt_lon": 52.527128,
    "shape_pt_sequence": 152,
  },
]
`;
